doc/arch/adr-002-delete-unpublished-documents-from-index.md

```diff
@@ -0,0 +1,90 @@
+# Decision record: delete documents from the govuk search index when an unpublishing message is received
```

Comment:
Yeah, when I wrote the title I was thinking of the ADR as just covering the decision to delete unpublished docs rather than keeping them in the index, but it actually covers all of the versioning decisions. Is the new name clearer?